pandas/core/indexes/base.py

```diff
@@ -3876,7 +3876,10 @@ def _join_monotonic(self, other, how="left", return_indexers=False):
             return join_index
 
     def _wrap_joined_index(self, joined, other):
-        name = get_op_result_name(self, other)
+        if isinstance(self, ABCMultiIndex):
+            name = self.names if self.names == other.names else None
+        else:
+            name = get_op_result_name(self, other)
```

Comment:
hmm why can't we do this in get_op_result_name?

Reply:
i considered that, decided against it because it meant additional overhead in a _lot_ of places

Reply:
kk